src/storage/src/hummock/sstable/mod.rs

```diff
@@ -180,20 +183,26 @@ impl SstableMeta {
     /// ```
     pub fn encode_to_bytes(&self) -> Vec<u8> {
         let mut buf = Vec::with_capacity(DEFAULT_META_BUFFER_CAPACITY);
+        self.encode_to(&mut buf);
+        buf
+    }
+
+    pub fn encode_to(&self, buf: &mut Vec<u8>) {
+        let start_offset = buf.len();
         buf.put_u32_le(self.block_metas.len() as u32);
         for block_meta in &self.block_metas {
-            block_meta.encode(&mut buf);
+            block_meta.encode(buf);
         }
-        put_length_prefixed_slice(&mut buf, &self.bloom_filter);
+        put_length_prefixed_slice(buf, &self.bloom_filter);
         buf.put_u32_le(self.estimated_size as u32);
         buf.put_u32_le(self.key_count as u32);
-        put_length_prefixed_slice(&mut buf, &self.smallest_key);
-        put_length_prefixed_slice(&mut buf, &self.largest_key);
-        let checksum = xxhash64_checksum(&buf);
+        put_length_prefixed_slice(buf, &self.smallest_key);
```

Comment:
how about use a struct called `Footer` to maintain this infomation (MAGIC, CHECKSUM, VERSION..)? Because , the `footer` for this PR only means `meta_offset`, and it not be placed at the end of `sstable` as a `footer`.

IIUC, we can use a fixed-length `Footer` struct and read it directly when we only need the `Footer` ?

Another way , aligned to the outside named it `meta_offset` ?

Reply:
You are right. It's not footer because it is not the last 8 bytes in this object.
Fixed.